hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -837,10 +840,75 @@ public void update(HoodieCleanMetadata cleanMetadata, String instantTime) {
    */
   @Override
   public void update(HoodieRestoreMetadata restoreMetadata, String instantTime) {
-    processAndCommit(instantTime, () -> HoodieTableMetadataUtil.convertMetadataToRecords(engineContext,
-        metadataMetaClient.getActiveTimeline(), restoreMetadata, getRecordsGenerationParams(), instantTime,
-        metadata.getSyncedInstantTime()), false);
-    closeInternal();
+    dataMetaClient.reloadActiveTimeline();
+
+    // Since the restore has completed on the dataset, the latest write timeline instant is the one to which the
+    // restore was performed. This should be always present.
+    final String restoreToInstantTime = dataMetaClient.getActiveTimeline().getWriteTimeline()
+        .getReverseOrderedInstants().findFirst().get().getTimestamp();
+
+    // We cannot restore to before the oldest compaction on MDT as we don't have the basefiles before that time.
+    Option<HoodieInstant> lastCompaction = metadataMetaClient.getCommitTimeline().filterCompletedInstants().lastInstant();
```

Comment:
shouldn't this logic go into restore method in BaseHoodieWriteClient where we trigger the restore for data table. so, this is double/repeated validation right?

Reply:
Yes, the BaseHoodieWriteClient also has this check. It is duplicated as we allow update() methods on the MDT to be called from outside. Its safer this way I suppose though duplicated.

Reply:
is it oldest Compaction or latest Compaction. java docs does not align w/ code.

Reply:
>  Its safer this way I suppose though duplicated

The code is hard to maintain, at least we abstract some utilities for these duplicate checks.

Reply:
Renamed to oldestCompaction and fixed the retrieval.